vaas/vaas/router/api.py

```diff
@@ -11,13 +12,59 @@
 
 from vaas.external.api import ExtendedDjangoAuthorization as DjangoAuthorization
 from vaas.external.serializer import PrettyJSONSerializer
-from vaas.router.models import Route, PositiveUrl, provide_route_configuration
+from vaas.router.models import Route, PositiveUrl, Rewrite, RewritePositiveUrl, provide_route_configuration
 from vaas.router.forms import RouteModelForm
 from vaas.router.report import fetch_urls_async, prepare_report_from_task
 from vaas.adminext.widgets import split_complex_condition, split_condition
 from vaas.external.oauth import VaasMultiAuthentication
 
 
+class RewritePositiveUrlResource(Resource):
+    url = fields.CharField(attribute='url')
+    expected_location = fields.CharField(attribute='expected_location')
+
+    def dehydrate(self, bundle):
+        bundle = super().dehydrate(bundle)
+        del bundle.data['resource_uri']
+        return bundle
+
+
+class RewriteResource(ModelResource):
+    rewrite_positive_urls = fields.ToManyField('vaas.router.api.RewritePositiveUrlResource', 'rewrite_positive_urls', full=True)
+
+    class Meta:
+        queryset = Rewrite.objects.all().prefetch_related('rewrite_positive_urls')
+        resource_name = 'rewrite'
+        serializer = PrettyJSONSerializer()
+        authorization = DjangoAuthorization()
+        authentication = VaasMultiAuthentication(ApiKeyAuthentication())
+        always_return_data = True
+
+    def save(self, bundle, *args, **kwargs):
+        rewrite_positive_urls = bundle.data.get('rewrite_positive_urls', [])
+        bundle.data['rewrite_positive_urls'] = []
+        bundle = super().save(bundle, *args, **kwargs)
+        if len(rewrite_positive_urls) != 0:
+            # if PATCH request not contains rewrite_positive_urls field we do not want to deleting existing rewrite_positive_urls
+            if not self._is_patch_request_without_rewrite_positive_urls_field(rewrite_positive_urls):
+                # if HTTP request contains rewrite_positive_urls we want delete previous rewrite_positive_urls attached to the Rewrite object
+                bundle.obj.rewrite_positive_urls.all().delete()
+                for rewrite_positive_url in rewrite_positive_urls:
+                    RewritePositiveUrl.objects.create(
```

Comment:
maybe we can use here [bulk_create()](https://docs.djangoproject.com/en/4.1/ref/models/querysets/#bulk-create), but only if we don't use post_save signals because they are not fired in bulk operations

Reply:
separate task